src/rpc/metrics_layer.rs

```diff
@@ -44,7 +44,8 @@ where
 
             metrics::RPC_METHOD_TIME
                 .get_or_create(&method)
-                .observe(start_time.elapsed().as_secs_f64());
+                // Observe the elapsed time in milliseconds
+                .observe(start_time.elapsed().as_secs_f64() * 1000.0);
```

Comment:
Too bad `.as_millis_f64()` isn't stable.

Reply:
Yeah, that is disappointing.

Reply:
Could still use a `millisecond` constant from Duration instead of the magic number.
```suggestion
                .observe(start_time.elapsed().as_secs_f64() * Duration::MILLISECOND as f64);
```

Reply:
No, we couldn't. That constant is nightly-only. https://doc.rust-lang.org/std/time/struct.Duration.html#associatedconstant.MILLISECOND

Even if it wasn't nightly, the suggested code would still not work. Conceptually, you'd convert, e.g., 3s to 3ms.

We could introduce a named constant, but given it's used exclusively here and it's pretty trivial, combined with the comment above, I'm not convinced it'd bring any value.

Do you agree?

Reply:
Oops, yeah, this was a bit rushed. Not a blocker, however still not a fan of magic numbers.